src/Lecture3.hs

```diff
@@ -211,10 +249,26 @@ types that can have such an instance.
 -}
 
 -- instance Foldable Weekday where
+--   impossible since Weekday has kind *
 -- instance Foldable Gold where
+--   impossible since Gold has kind *
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+--   impossible since Reward has kind *
+instance Foldable List1 where
+  foldr :: (a -> b -> b) -> b -> List1 a -> b
+  foldr alg acc (List1 a []) = a `alg` acc
+  foldr alg acc (List1 a (x:xs)) = a `alg` (foldr alg acc (List1 x xs))
+
+  foldMap :: Monoid m => (a -> m) -> List1 a -> m
+  -- foldMap f (List1 a []) = f a
+  -- foldMap f (List1 a (x:xs)) = f a <> foldMap f (List1 x xs)
+  foldMap f = foldr (\a b -> f a <> b) mempty
```

Comment:
This is the default implementation of `foldMap` via `foldr` so in this form there's no much benefit in writing it explicitly. You can delegate to `foldMap` for ordinary lists (and you can do the same for `List1` as well to avoid allocating extra lists:

```haskell
foldMap f (List1 x xs) = f x <> foldMap f xs
```